databricks/koalas/series.py

```diff
@@ -4283,14 +4283,23 @@ def replace(self, to_replace=None, value=None, regex=False) -> "Series":
                 current = self.spark.column
             else:
                 for to_replace_, value in to_replace.items():
+                    cond = (
+                        (F.isnan(self.spark.column) | self.spark.column.isNull())
```

Comment:
Not pretty sure, but I think maybe we don't need `F.isnan(self.spark.column)` because both `None` and `np.nan` cast to `null` in the internal Spark DataFrame ??

```python
>>> kser_with_nan
0    1.0
1    2.0
2    NaN
3    4.0
dtype: float64

>>> kser_with_none
0       a
1       b
2    None
3       d
dtype: object

>>> kser_with_nan._internal.spark_frame.select(kser_with_nan.spark.column).show()
+----+
|   0|
+----+
| 1.0|
| 2.0|
|null|
| 4.0|
+----+

>>> kser_with_none._internal.spark_frame.select(kser_with_none.spark.column).show()
+----+
|   0|
+----+
|   a|
|   b|
|null|
|   d|
+----+
```

Reply:
I think we can keep using `F.isnan`, some use cases may produce NaN in the internal Spark DataFrame : 

```python
>>> kdf = ks.DataFrame({"A": [-1, 2, 3, 4, np.nan]})
>>> kdf._internal.spark_frame.withColumn("sqrt(A)", sqrt("A")).show()
+------------------+
|           sqrt(A)|
+------------------+
|               NaN|
|1.4142135623730951|
|1.7320508075688772|
|               2.0|
|              null|
+------------------+
```

Reply:
Great! let's keep it 👍